File: pdu/python/std_msgs/pdu_cdr_conv_UInt32MultiArray.py
from .pdu_pytype_UInt32MultiArray import UInt32MultiArray
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..std_msgs.pdu_cdr_conv_MultiArrayDimension import *
from ..std_msgs.pdu_cdr_conv_MultiArrayLayout import *



def py_to_cdr_body_UInt32MultiArray(writer: CdrWriter, src: UInt32MultiArray):
    py_to_cdr_body_MultiArrayLayout(writer, src.layout)
    writer.write_sequence_length(src.data)
    for elem in src.data:
        writer.write_uint32(elem)


def cdr_body_to_py_UInt32MultiArray(reader: CdrReader, dst: UInt32MultiArray):
    cdr_body_to_py_MultiArrayLayout(reader, dst.layout)
    dst.data = [reader.read_uint32() for _ in range(reader.read_uint32())]
    return dst


def py_to_cdr_UInt32MultiArray(src: UInt32MultiArray) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_UInt32MultiArray(writer, src)
    return writer.bytes()


def cdr_to_py_UInt32MultiArray(cdr_payload) -> UInt32MultiArray:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = UInt32MultiArray()
    return cdr_body_to_py_UInt32MultiArray(reader, dst)

File: pdu/javascript/std_msgs/pdu_cdr_conv_UInt32MultiArray.js
import { UInt32MultiArray } from './pdu_jstype_UInt32MultiArray.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { MultiArrayDimension } from '../std_msgs/pdu_jstype_MultiArrayDimension.js';
import { PduMultiArrayDimensionConverter } from '../std_msgs/pdu_cdr_conv_MultiArrayDimension.js';
import { MultiArrayLayout } from '../std_msgs/pdu_jstype_MultiArrayLayout.js';
import { PduMultiArrayLayoutConverter } from '../std_msgs/pdu_cdr_conv_MultiArrayLayout.js';


export class PduUInt32MultiArrayConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { UInt32MultiArray } src
     */
    static to_cdr_body(writer, src) {
        PduMultiArrayLayoutConverter.to_cdr_body(writer, src.layout);
        writer.write_sequence_length(src.data);
        for (const elem of src.data) {
            writer.write_uint32(elem);
        }
    }

    /**
     * @param {PduCdrReader} reader
     * @param { UInt32MultiArray } dst
     * @returns { UInt32MultiArray }
     */
    static cdr_body_to_js(reader, dst) {
        PduMultiArrayLayoutConverter.cdr_body_to_js(reader, dst.layout);
        dst.data = Array.from({ length: reader.read_uint32() }, () => reader.read_uint32());
        return dst;
    }

    /**
     * @param { UInt32MultiArray } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { UInt32MultiArray }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new UInt32MultiArray());
    }
}
